include/alpaka/dev/DevGenericSycl.hpp

```diff
@@ -135,17 +135,27 @@ namespace alpaka::experimental
             return std::make_pair(m_impl->get_device(), m_impl->get_context());
         }
 
-        std::shared_ptr<detail::DevGenericSyclImpl> m_impl;
+        auto get_device() const -> sycl::device
+        {
```

Comment:
Should we call those maybe `get_native_device`? We tend to us the word `native` when we expose platform specifics. See also `getNativeHandle`, which btw already offers this functionality. Should we have your function at all then?

Reply:
The two functions get_device and get_context were implemented as commodities for other pieces of code more than anything. In particular, sometimes we only need one of the two (for example, to allocate pinned memory on host we just need the device's context) and I find it clearer to ask for either the device or the context with a specific function instead of relying on the pair. I might reimplement them using getNativeHandle and maybe rename them, following the same convention, as getNativeDevice and getNativeContext. Otherwise, removing them entirely shouldn't be much of an issue

Reply:
There are a few cases where we need the context but not the device: the allocation of pinned / mapped host memory. 
I'll look into all the debug prints (also, in the `sycl::free` method we only need the context)